src/read/unit.rs

```diff
@@ -2279,6 +2306,115 @@ impl<'abbrev, 'entry, 'unit, R: Reader> FallibleIterator for AttrsIter<'abbrev,
     }
 }
 
+/// A raw reader of the data that defines the Debugging Information Entries.
+///
+/// This reader provides primitives to read the components of Debugging Information
+/// Entries (DIEs). A DIE consists of an abbreviation code (read with `read_abbreviation`)
+/// followed by a number of attributes (read with `read_attribute`).
+/// The user must provide the control flow to read these correctly.
+/// In particular, all attributes must always be read before reading another
+/// abbreviation code.
```

Comment:
It would be nice to add a section about "why would I use the raw interface instead of the cursor/tree?" to these docs.